README.md

```diff
@@ -0,0 +1,38 @@
+## node reference
+
+Outline transport agnostic node js service. 
+
+
+directory layout 
+
+```sh
+.
+├── config  
+├── context # generated contexts generated each time
+├── deploy # our business logic not generated 
+└── endpoints
+    └── http #generated endpoints. Each endpoint only generated once    
+
+```
+
+
+files layout 
+
+```sh 
+├── config
+│   ├── configValidation.js
+│   └── dev.json
+├── context
+│   └── deploy.js
+├── deploy # directory named after the domain the business logic covers 
```

Comment:
I don't mind doing this but then your just hiding the messiness one layer down. I kind of like that the business domains of the app are clearly visible at the top layer, however this is definitely not a big deal